modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -3289,6 +3296,34 @@ protected final boolean hasValueUnlocked() {
         return val != null;
     }
 
+    /**
+     * Checks, that changes were got by DR.
+     *
+     * @param explicitVer – Explicit version (if any).
+     * @return {@code true} if changes were got by DR and {@code false} otherwise.
+     */
+    private boolean isRemoteDrUpdate(@Nullable GridCacheVersion explicitVer) {
+        return explicitVer != null && explicitVer.dataCenterId() != cctx.dr().dataCenterId();
+    }
+
+    /**
+     * Checks, that cache interceptor should be skipped.
+     * <p>
+     * In current cache interceptor implementation, if changes were got by DR, the following methods will be invoked:
+     * {@link CacheInterceptor#onBeforePut(Cache.Entry, Object)}, {@link CacheInterceptor#onAfterPut(Cache.Entry)},
+     * {@link CacheInterceptor#onBeforeRemove(Cache.Entry)} and {@link CacheInterceptor#onAfterRemove(Cache.Entry)}, but
+     * method {@link CacheInterceptor#onGet(Object, Object)} will not be invoked. This implementation could corrupt
+     * data. For example, cache interceptor wrap data on update/remove and unwrap data back on get from cache. For avoid
```

Comment:
Let's fix use case description to smth like this:

It is expects by default behavior that  mentioned Interceptor methods will be called, but onGet(). This can even make DR-update flow broken in case of non-idempotent Interceptor and force users to call onGet manually as the only workaround.
Also, user may want to skip Interceptor to avoid redundant entry transformation for DR updates and exchange with internal data b/w data centres which is a normal case.